sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -862,4 +915,85 @@ mod checked {
             pt,
         )
     }
+
+    fn setup_authenticator_state_create(
+        mut builder: ProgrammableTransactionBuilder,
+        _create: AuthenticatorStateCreate,
+    ) -> ProgrammableTransactionBuilder {
+        builder
+            .move_call(
+                SUI_FRAMEWORK_ADDRESS.into(),
+                AUTHENTICATOR_STATE_MODULE_NAME.to_owned(),
+                AUTHENTICATOR_STATE_CREATE_FUNCTION_NAME.to_owned(),
+                vec![],
+                vec![],
+            )
+            .expect("Unable to generate authenticator_state_create transaction!");
+        builder
+    }
+
+    fn setup_authenticator_state_update(
+        update: AuthenticatorStateUpdate,
+        temporary_store: &mut TemporaryStore<'_>,
+        tx_ctx: &mut TxContext,
+        move_vm: &Arc<MoveVM>,
+        gas_charger: &mut GasCharger,
+        protocol_config: &ProtocolConfig,
+        metrics: Arc<LimitsMetrics>,
+    ) -> Result<(), ExecutionError> {
+        let pt = {
+            let mut builder = ProgrammableTransactionBuilder::new();
+            let res = builder.move_call(
+                SUI_FRAMEWORK_ADDRESS.into(),
+                AUTHENTICATOR_STATE_MODULE_NAME.to_owned(),
+                AUTHENTICATOR_STATE_UPDATE_FUNCTION_NAME.to_owned(),
+                vec![],
+                vec![
+                    CallArg::Object(ObjectArg::SharedObject {
+                        id: SUI_AUTHENTICATOR_STATE_OBJECT_ID,
+                        initial_shared_version: SUI_AUTHENTICATOR_STATE_OBJECT_SHARED_VERSION,
```

Comment:
What is this?? How does it pass tests?

Reply:
Leftover from before the EndOfEpochTransaction refactor, thanks for the catch. It passes because as far as I know, initial_shared_version is not used by the runtime, so you can pass in anything you want here. Fixing.